make/tools.mk

```diff
@@ -482,8 +512,31 @@ tidy:
 .PHONY: go-workspace
 go-workspace: export GOWORK?=$(abspath go.work)
 ## Create a go.work file in the repository root (or GOWORK)
+##
 ## @category Development
 go-workspace:
 	@rm -f $(GOWORK)
 	go work init
 	go work use . ./cmd/acmesolver ./cmd/cainjector ./cmd/controller ./cmd/ctl ./cmd/webhook ./test/integration ./test/e2e
+
+.PHONY: learn-sha-tools
+## Re-download all tools and update the tools.mk file with the
+## sha256sums of the downloaded tools. This is useful when you
+## update the version of a tool in the Makefile, and want to
+## automatically update the sha256sums in the tools.mk file.
+##
+## @category Development
+learn-sha-tools:
+	rm -rf ./$(BINDIR)
+	mkdir ./$(BINDIR)
+	$(eval export LEARN_FILE=$(PWD)/$(BINDIR)/learn_file)
```

Comment:
By setting the `LEARN_FILE` environment variable, ./hack/util/checkhash.sh will write a "replace hash x with y" command to the learn file instead of throwing an error on mismatch.

At the end of this make target, we use this learn file to update the tools.mk file and replace all the old hashes with the desired new hashes.